### PR TITLE
ci: Use devcontainer

### PR DESCRIPTION
**Pull request by olix0r**
Signed-off-by: Oliver Gould <ver@buoyant.io>

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,13 +1,13 @@
 {
     "name": "linkerd2-proxy-api",
-    "image": "ghcr.io/linkerd/dev:v15",
+    "image": "ghcr.io/linkerd/dev:v22",
     "extensions": [
         "DavidAnson.vscode-markdownlint",
         "golang.go",
+        "kokakiwi.vscode-just",
```
Comment (hawkw) [inline .devcontainer/devcontainer.json:7]: oh, cool, this one has syntax highlighting that actually works!

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -21,18 +21,16 @@ jobs:
...
         run: |
-          ref=${{ github.ref }}
-          if [[ "$ref" == refs/tags/release/* ]]; then
+          ref='${{ github.ref }}'
+          if [[ "$ref" =~ ^refs/tags/v[0-9]+\.[0-9]+\.[0-9]+ ]]; then
             echo ::set-output name=publish::true
             echo ::set-output name=version::"${ref##refs/tags/release/}"
           else
-            sha=${{ github.sha }}
+            #sha=${{ github.sha }}
```
Comment (hawkw) [inline .github/workflows/release.yml:33]: is this supposed to be commented out? should we just remove it (and remove the reference to `sha` on the next line)?

**Review by hawkw — COMMENTED** [review]
one question about the commented-out line of bash; looks good besides that!

File: .github/workflows/release.yml (hunk 2 above)
Comment (olix0r) [reply .github/workflows/release.yml:33]: good catch. shouldn't be commented

**Pull request merged**